core/domain/state_domain_test.py

```diff
@@ -458,13 +458,20 @@ def test_hints_validation(self):
         exploration.validate()
 
         # Add hint and delete hint.
-        init_state.add_hint(state_domain.SubtitledHtml('hint_2', 'new hint'))
+        init_state.interaction.hints.append(
+            state_domain.Hint(
+                state_domain.SubtitledHtml('hint_2', 'new hint')))
         self.assertEqual(
             init_state.interaction.hints[1].hint_content.html,
             'new hint')
-        init_state.add_hint(
-            state_domain.SubtitledHtml('hint_3', 'hint three'))
-        init_state.delete_hint(1)
+        init_state.interaction.hints.append(
+            state_domain.Hint(
+                state_domain.SubtitledHtml('hint_3', 'hint three')))
+
+        if len(init_state.interaction.hints) <= 1:
```

Comment:
Why this check required in the test file?

Reply:
For the file for now... No, but I thought maybe if the file is modified by someone, just for a precaution.
Will remove it for now.
Thanks!